File: GenesAndGenealogy/GenesAndGenealogy.Server/ViewModels/MultimediaModel.cs
using Gedcom.RecordStructures;

namespace GenesAndGenealogy.Server.ViewModels;

public class MultimediaModel(MultimediaRecord multimediaRecord)
{
    public string? AutomatedRecordId { get; set; } = multimediaRecord.AutomatedRecordId;
    public ChangeDate? ChangeDate { get; set; } = multimediaRecord.ChangeDate;
    public string? Date { get; set; } = multimediaRecord.Date;
    public string? Description { get; set; } = multimediaRecord.Description;
    public string? DescriptiveTitle { get; set; } = multimediaRecord.DescriptiveTitle;
    public FileRecord? FileRecord { get; set; } = multimediaRecord.FileRecord;
    public List<string>? MultimediaFileReferenceNumbers { get; set; } = multimediaRecord.MultimediaFileReferenceNumbers;
    public MultimediaFormat? MultimediaFormat { get; set; } = multimediaRecord.MultimediaFormat;
    public List<NoteStructure>? Notes { get; set; } = multimediaRecord.NoteStructures;
    public string? ObjectId { get; set; } = multimediaRecord.ObjectId;
    public PlaceStructure? Place { get; set; } = multimediaRecord.PlaceStructure;
    public List<SourceCitation>? SourceCitations { get; set; } = multimediaRecord.SourceCitations;
    public UserReferenceNumber? UserReferenceNumber { get; set; } = multimediaRecord.UserReferenceNumber;

    public override string ToString() => $"{Description} ({DescriptiveTitle})";
}

File: GenesAndGenealogy/genesandgenealogy.client/src/view-models/MultimediaModel.ts
import { FileModel } from './FileModel';
import { NoteModel } from './NoteModel';
import { PlaceModel } from './PlaceModel';

export interface MultimediaModel {
  automatedRecordId: string;
  //ChangeDate: ChangeDateJson;
  date: string;
  description: string;
  descriptiveTitle: string;
  file: FileModel;
  multimediaFileReferenceNumbers: string[];
  //MultimediaFormat: MultimediaFormatJson;
  notes: NoteModel[];
  objectId: string;
  place: PlaceModel;
  //sourceCitations: SourceCitationJson[];
  //userReferenceNumber: UserReferenceNumberJson;
}

